exercises/exercise5/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Node struct {
+	Value int
+	Next  *Node
+}
+
+func Prepend(head **Node, x int) {
```

Comment:
Please consider using idiomatic Go:
```go
func Prepend(head **Node, x int) {
  *head = &Node{Value: x, Next: *head}
}
```

Reply:
Wait, is this (`&Node{Value: x, Next: *head}`) object heap allocated too?

Reply:
Golang handles heap/stack automatically. This is one of the things the language is built around. It will detect the node leaks the scope and will heap-allocate it. Yes, in the same way as `new` does.